src/Models/User.php

```diff
@@ -105,14 +105,14 @@ public function getFullNameAttribute()
         return "{$this->name} {$this->name_last}";
     }
 
-    public function toCustomer(array $attributes): Customer
+    public function toCustomer(array $attributes)
     {
         $this->assignRole('Customer');
 
         return $this->customers()->create($attributes);
     }
 
-    /**
+    /*/**
      * Get active cart.
      *
      * @return Cart
```

Comment:
I have to comment the `@return Cart` because is the only place in the `Authorization` package that require the model `Cart` that belongs to `Checkout` package... what is the best option to handle this @sl0wik @pdbreen

Reply:
Should be replace with something like this.  Return value will also change to `CartInterface` (which is defined in support) instead of Cart. 

```
if ($service = findService(CartInterface::class)) {
return $service::activeCart($this-id);
}

return null;
```

Reply:
Beautiful. 🤩

Reply:
@pdbreen I tried with the approach you gave me but the findService retrieve me an error ` Method Tipoff\Support\Contracts\Services\BaseService::activecart does not exist `

instead I made this and works. The activeCart method create a new one if does not exist... so always return the activeCart... Let me know if this ok

        /** @var CartInterface $cartInterface */
        $cartInterface = app(CartInterface::class);
        return $cartInterface::activeCart($this->id);

Reply:
That will throw an exception if the service doesn’t exist.   Better to Typehint the response from findService instead and handle the possibility of null response

Reply:
@pdbreen  ok to handle the responsability to return null I have to remove this below

    /**
     * Get active cart.
     *
     * @return CartInterface
     */

![image](https://user-images.githubusercontent.com/13080491/109395663-919c5c00-78fb-11eb-8abd-853e93852204.png)


The other error I told you is 

        if ($service = findService(CartInterface::class)) {
            return $service::activeCart($this->id);
        }

![image](https://user-images.githubusercontent.com/13080491/109395694-bb558300-78fb-11eb-9e6e-cd09828c0ae4.png)

Reply:
‘@return CartInterface|null’ and add type hint to $service as CartInterface - same as you did when using ‘app(...)’

Reply:
@pdbreen CartInterface|null’ works. Thanks

/** @var CartInterface $service */
if ($service = findService(CartInterface::class)) .....

I'm receiving the same issue, I dont know if I missing something

Reply:
Psalm is weird.  Move that comment just underneath the `if (..)` and it will work.

Reply:
Thanks that work